test/jdk/java/awt/FullScreen/FullScreenInsets/FullScreenInsets.java

```diff
@@ -21,45 +21,53 @@
  * questions.
  */
 
+import javax.imageio.ImageIO;
 import java.awt.AWTException;
 import java.awt.Color;
 import java.awt.Dimension;
-import java.awt.DisplayMode;
 import java.awt.Frame;
 import java.awt.GraphicsDevice;
 import java.awt.GraphicsEnvironment;
 import java.awt.Insets;
+import java.awt.Rectangle;
 import java.awt.Robot;
 import java.awt.Window;
 import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
 
-/**
+/*
  * @test
  * @key headful
  * @bug 8003173 7019055
  * @summary Full-screen windows should have the proper insets.
- * @author Sergey Bylokhov
+ * @run main FullScreenInsets
  */
 public final class FullScreenInsets {
 
     private static boolean passed = true;
     private static Robot robot = null;
+    private static int deviceCount = 0;
+    private static final float TOLERANCE = 10;
 
-    public static void main(final String[] args) {
+    public static void main(final String[] args) throws IOException {
         final GraphicsEnvironment ge = GraphicsEnvironment
                 .getLocalGraphicsEnvironment();
         final GraphicsDevice[] devices = ge.getScreenDevices();
+        System.out.println("No. of Screen Devices: "+ devices.length + "\n");
 
-        final Window wGreen = new Frame();
+        final Frame wGreen = new Frame();
         wGreen.setBackground(Color.GREEN);
+        wGreen.setUndecorated(true);
```

Comment:
Why did you make that frame undecorated? I think the decoration was one of the reasons why the bug was reproduced.

Reply:
When testing on Win10, observed that for a decorated frame the title bar is retained in full screen mode unlike some of the other applications (eg. Chrome and IDE in full screen mode). 

When I checked the Javadocs for [setFullScreenWindow(Window w)](https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/java/awt/GraphicsDevice.html#setFullScreenWindow(java.awt.Window)), it states that - _"It is unspecified and platform-dependent how decorated windows operate in full-screen mode. For this reason, it is recommended to turn off the decorations in a Frame or Dialog object by using the setUndecorated method."_  

For the above reasons and to aid testing the pixel-color across full-screen window without having to account for different platform specific title bar sizes, the frame was set as undecorated.

Reply:
You just uncover this bug https://bugs.openjdk.org/browse/JDK-7019055 referenced in the test, and one of the reasons why the specification is state what it states. SO it is better to not make the frame decorated but instead exclude it based on the current graphics device, OGL/D3D vs win32(or something like that)